packages/core/src/guards.ts

```diff
@@ -138,6 +138,32 @@ function checkNot(
   return !evaluateGuard(guards[0], context, event, snapshot);
 }
 
+/**
+ * Higher-order guard that evaluates to `true` if the `guard` passed to it evaluates to `false`.
+ *
+ * @category Guards
+ * @example
+  ```ts
+  import { setup, not } from 'xstate';
+
+  const machine = setup({
+    guards: {
+      someNamedGuard: () => false
+    }
+  }).createMachine({
+    on: {
+      someEvent: {
+        guard: not('someNamedGuard'),
+        actions: () => {
+          // will be executed if guard in `not(...)`
+          // evaluates to `false`
+        }
+      }
+    }
+  });
+  ```
+ * @returns A guard action object
```

Comment:
the return value is not exactly a plain object (it's a function)

Reply:
For the developer's intents, it is an object (saying it's a function would be confusing, and is an implementation detail)

Reply:
"guard action" is definitely confusing, we can just make it "guard". Similarly, we don't even need to specify what it is - for the developer's intent it's just a guard (a "guard thing" but calling it a "guard thing" would be silly :D), they shouldn't inspect it anyhow so the extra "object" bit isn't that helpful here anyway